### ISSUE TITLE
Bing.com images outrageous memory usage

### ISSUE THREAD

**Issue by hockeymikey** [BrowserWorks/waterfox#916]
**Describe the bug**
Looking at images on bing.com causes an uncontrollable memory leak/expansion that never stops until I close the tab/navigate away/ect.

**To Reproduce**
1. Go to 'bing.com'
2. Click on 'search for something and click on images'
3. Watch your memory, gets 

**Expected behavior**
No outrageous memory usage

**Desktop (please complete the following information):**
 - OS: Windows 7
 - Version 56.2.8

**Comment by grahamperrin** [comment]
Reproducible with Waterfox in safe mode? 

- about:support

Reproducible with a new profile? 

- about:profiles

No problem here with 56.2.8 on FreeBSD-CURRENT, https://www.bing.com/images/search?q=memory+usage&go=Search loaded reasonably quickly and memory usage is OK, not rising.

**Comment by InI4** [comment]
Vaguely connected remark/rant: Firefox (and therefore Waterfox) massively suffers from memory leaks for me as a user. 

I have filed many bugs with them, as it is daily reproducible for me in my 40 Tabs environment with a "never close before it has 6GB RAM and is really slow"-policy with the addons I have. However they have no clue how to research this and are usually priorizing these cases down (I am not the only creator). They only find, I have so many ghostwindows, which seem to eat both memory and CPU in form of garbage collection. 

I am allmost sure, it has to do with certain JavaScript/Ajax stuff and essentially Google/Twitter sites. When I tested, before moving to Waterfox, Waterfox always fared better than corresponding Firefox, unfortunately only slightly.

**Comment by tcreek** [comment]
I agree it has a very bad memory leak issue. On my Linux laptop, I normally just put the laptop to sleep instead of a shutdown. After a couple of weeks I notice it is performing very badly. I look at the memory usage and see Waterfox is the culprit of using nearly all 16GB of RAM on the system.

**Comment by quroe** [comment]
1451985 - Ghost windows from dropbox shared image link page
https://bugzilla.mozilla.org/show_bug.cgi?id=1451985
mozilla-esr60: changeset 449393:39832495babf
https://hg.mozilla.org/releases/mozilla-esr60/rev/39832495babf

related this?
Waterfox doesn't apply this,so some sites may cause leak.

**Comment by kazerniel** [comment]
I get a similar issue, where my memory usage climbs up slowly over time. My browser gets very sluggish at around 1.5 GB (I have multiprocess disabled), so I set the Memory Restart addon to run a minimize command when it reaches that point. If I restart the browser with the same tabs, it takes up just 250ish MB. (I probably don't get the same amount of leak as you guys get, because I restart my browser 1-2 times a day. I think the highest I got was 3GB from having multiple Twitch tabs open.)
